internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -7,26 +7,152 @@ import (
 	It("returns whether or not we're using NGINX Plus", func() {
-		mgr := NewManagerImpl(nil, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr := runtime.NewManagerImpl(nil, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeFalse())
 
-		mgr = NewManagerImpl(&ngxclient.NginxClient{}, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr = runtime.NewManagerImpl(&ngxclient.NginxClient{}, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeTrue())
 	})
+
+	var (
+		manager         runtime.Manager
+		upstreamServers []ngxclient.UpstreamServer
+		upstreamServer  ngxclient.UpstreamServer
+		ngxPlusClient   *runtimefakes.FakeNginxPlusClient
+		process         *runtimefakes.FakeProcessHandler
+
+		metrics      *runtimefakes.FakeMetricsCollector
+		verifyClient *runtimefakes.FakeVerifyClient
+	)
+
+	var (
+		MaxConnsMock = 150
+		MaxFailsMock = 20
+		WeightMock   = 10
+		BackupMock   = false
+		DownMock     = false
+	)
+
+	BeforeAll(func() {
+		upstreamServer = ngxclient.UpstreamServer{
+			ID:          1,
+			Server:      "10.0.0.1:80",
+			MaxConns:    &MaxConnsMock,
+			MaxFails:    &MaxFailsMock,
+			FailTimeout: "test",
+			SlowStart:   "test",
+			Route:       "test",
+			Backup:      &BackupMock,
+			Down:        &DownMock,
+			Drain:       false,
+			Weight:      &WeightMock,
+			Service:     "test",
+		}
+
+		upstreamServers = []ngxclient.UpstreamServer{
+			upstreamServer,
+		}
+
+	})
+
+	Context("Reload", func() {
+		BeforeAll(func() {
+			ngxPlusClient = &runtimefakes.FakeNginxPlusClient{}
+			process = &runtimefakes.FakeProcessHandler{}
+			metrics = &runtimefakes.FakeMetricsCollector{}
+			verifyClient = &runtimefakes.FakeVerifyClient{}
+			manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+		})
+
+		It("sucessfully reloads NGINX configuration", func() {
+			err := manager.Reload(context.Background(), 1)
+			Expect(err).To(BeNil())
+		})
+
+		When("not sucessfully reloads NGINX configuration", func() {
+			It("should panic if MetricsCollector not enabled", func() {
+				metrics = nil
+				manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+
+				reload := func() {
+					manager.Reload(context.Background(), 0)
+				}
+
+				Expect(reload).To(Panic())
+			})
+
+			It("should panic if VerifyClient not enabled", func() {
+				metrics = &runtimefakes.FakeMetricsCollector{}
+				verifyClient = nil
+				manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+
+				reload := func() {
+					manager.Reload(context.Background(), 0)
+				}
+
+				Expect(reload).To(Panic())
+			})
+		})
+	})
+
+	When("running NGINX plus", func() {
+		BeforeAll(func() {
+			ngxPlusClient = &runtimefakes.FakeNginxPlusClient{}
+			manager = runtime.NewManagerImpl(ngxPlusClient, nil, zap.New(), nil, nil)
+		})
+
+		It("sucessfully updates HTTP server upstream", func() {
+			err := manager.UpdateHTTPServers("test", upstreamServers)
+
+			Expect(err).To(BeNil())
```

Comment:
```suggestion
			Expect(manager.UpdateHTTPServers("test", upstreamServers)).To(Succeed())
```

Reply:
thanks for suggestion Im going to add an update